runtime/reconcilers/model.go

```diff
@@ -276,6 +281,13 @@ func (r *ModelReconciler) Reconcile(ctx context.Context, n *runtimev1.ResourceNa
 		return runtime.ReconcileResult{Err: ctx.Err()}
 	}
 
+	// Open executor for the new output
+	executorConnector, executor, release, err := r.acquireExecutor(ctx, opts)
+	if err != nil {
+		return runtime.ReconcileResult{Err: err}
+	}
+	defer release()
+
 	// Build the output
 	execRes, execErr := executor.Execute(ctx)
```

Comment:
Maybe combine these two steps into a `r.execute` function that handles the staging and non-staging cases?